form.php

```diff
@@ -0,0 +1,96 @@
+
+
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>Form</title>
+    </head>
+    <body>
+        <?php
+            $error1 = [];
+
+            class user_validate
+            {
+                private $name,$val;
+                public $error=['first'=>'', 'last'=>'', 'full'=>'', 'out'=>''];
+
+
+                private static $field = ['first', 'last'];
+
+                public function __construct($post_data)
+                {
+                    $this->name = $post_data;
+                }
+
+                public function validateForm()
+                {
+                    foreach(self::$field as $value)
+                    {
+                        if(empty($this->name[$value]))
+                        {
+                            $this->$error[$value] = "Required";
+                        }
+                        else
+                        {
+                            $n = $this->validate_str($this->name[$value]);
+
+                            if (!preg_match("/^[a-zA-Z- ]*$/", $n))
+                            {
+                                $this->$error[$value] = "Characters Only";
+                            }
+
+
+                        }
+                    }
+
+                    if (!($this->$error['first'] == 'Required') && !($this->$error['last'] == 'Required'))
+                    {
+                        $this->$error['full'] = $this->name['first'] . " " . $this->name['last'];
```

Comment:
If `$this->error` is storing the correct output then why is it called `error`. Use proper variable name.